File: backend/src/schemas/partner.py
# src/schemas/partner.py
from pydantic import BaseModel, confloat
from typing import List, Optional
from datetime import datetime
from enum import Enum
from .enums import PartnerRole, BookingType

class PartnerFilter(BaseModel):
    role: Optional[PartnerRole] = None
    min_rating: Optional[float] = None
    min_experience: Optional[int] = None 
    max_hourly_rate: Optional[float] = None
    specialization: Optional[str] = None

class PartnerPricing(BaseModel):
    hourly_rate: confloat(ge=0)
    daily_rate: confloat(ge=0)
    monthly_rate: confloat(ge=0)

class PartnerResponse(BaseModel):
    id: int
    full_name: str
    role: PartnerRole
    experience_years: int
    rating: float
    total_reviews: int
    specializations: List[str]
    pricing: PartnerPricing
    kecamatan: str
    is_available: bool
    available_hours: Optional[List[dict]] = []
    profile_image: Optional[str] = None
    preferred_booking_type: Optional[BookingType] = None
    languages: Optional[List[str]] = []
    profile_description: Optional[str] = None

    class Config:
        from_attributes = True

File: frontend/src/types/partner.ts
// src/types/partner.ts
import { MatchSortCriteria } from './enums'
export enum PartnerRole {
  PEMBANTU = 'pembantu',
  TUKANG_KEBUN = 'tukang_kebun',
  TUKANG_PIJAT = 'tukang_pijat',
}

export enum BookingType {
  HOURLY = 'hourly',
  DAILY = 'daily',
  MONTHLY = 'monthly',
}

export interface PartnerPricing {
  hourly_rate: number
  daily_rate: number
  monthly_rate: number
}

export interface MatchRequest {
  start_datetime: string
  kecamatan: string
  role: PartnerRole
  booking_type: BookingType
  filters: {
    role: PartnerRole
    min_rating: number
    min_experience: number
    max_hourly_rate: number | null
    specialization: string | null
    kecamatan: string
  }
  sort_by: MatchSortCriteria
}

export interface PartnerFilter {
  role?: string
  min_rating?: number
  min_experience?: number
  max_hourly_rate?: number
  specialization?: string
}

// src/types/partner.ts
export interface Partner {
  id: number
  full_name: string
  role: 'pembantu' | 'tukang_kebun' | 'tukang_pijat'
  experience_years: number
  rating: number
  total_reviews: number
  specializations: string[]
  pricing: {
    hourly_rate: number
    daily_rate: number
    monthly_rate: number
  }
  kecamatan: string
  is_available: boolean
  profile_image?: string
  languages: string[]
  profile_description?: string
}
